added explaining concept of logical date in DAG run

howardyoo opened the pull request:
Added two paragraphs under the DAG run to describe the difference between logical date and dag start and end date, and some of the reason behind its naming 'logical,' as well as providing an example. The intention for this additional content is to clarify any confusions about what logical date would mean compared to the DAG run's start and end date.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

howardyoo committed:
added explaining concept of logical date in DAG run

Added two paragraphs under the DAG run to describe the difference between logical date and dag start and end date, and some of the reason behind its naming 'logical,' as well as providing an example.

howardyoo committed:
Update dags.rst

howardyoo committed:
Update dags.rst

changed  a bit of formatting

howardyoo committed:
added 'see also' section on three different areas of documentation to interlink each other

howardyoo committed:
Update dags.rst

howardyoo committed:
Update dag-run.rst

howardyoo committed:
Update faq.rst

potiuk reviewed (approved):
LGTM. @uranusjr ?

howardyoo committed:
Update faq.rst

fixed the incorrect typo

howardyoo committed:
Merge branch 'main' into patch-1

uranusjr commented on docs/apache-airflow/concepts/dags.rst:
```diff
@@ -177,6 +179,20 @@ In much the same way a DAG instantiates into a DAG Run every time it's run,
 :ref:`Task Instances <concepts:task-instances>` along with it.
 
+When the DAG run starts, it will have the start date and end date. This period
+describes the time when the DAG run actually 'ran.' Aside from the DAG
```
The DAG run does not have an end date until after it _finishes_, not when it starts. So the wording here probably needs to be tweaked slightly to avoid confusion.

howardyoo replied:
sure, makes sense - will apply the suggestion accordingly.

uranusjr commented on docs/apache-airflow/concepts/dags.rst:
```diff
@@ -177,6 +179,20 @@ In much the same way a DAG instantiates into a DAG Run every time it's run,
+(formally known as execution date), which describes the intended time for the
+DAG run to be scheduled or triggered to run. The reason why this is called
+``logical`` is because of the abstract nature of it having multiple meanings,
+depending on the context of the DAG run itself.
```
```suggestion
run's start and end date, there is another date called *logical date*
(formally known as execution date), which describes the intended time a
DAG run is scheduled or triggered. The reason why this is called
*logical* is because of the abstract nature of it having multiple meanings,
depending on the context of the DAG run itself.
```

Double backticks are for code, not emphasis. (same applies for other parts below that use backticks)

Also, perhaps it’s better to use present tense here? I feel future tense hints that this time is toward the _end_ of the data interval, which is not always the case. (Or maybe let’s remove the “which describes” part altogether? The most important takeaway here is that “logical date” is abstract and have multiple meanings, and everything else can be potentially misleading and dilute the thing we want to get across.)

uranusjr commented on docs/apache-airflow/concepts/dags.rst:
```diff
@@ -177,6 +179,20 @@ In much the same way a DAG instantiates into a DAG Run every time it's run,
+to DAG run's start date. However, when the DAG is being scheduled, with certain
+schedule interval put in place, this logical date is going to indicate the time
+at which it marks the start of the data interval, where the DAG run's start
+date would then be the logical date + scheduled interval.
```
```suggestion
For example, if a DAG run is manually triggered by the user, its logical date would be the
date and time of which the DAG run was triggered, and the value should be equal
to DAG run's start date. However, when the DAG is being automatically scheduled, with certain
schedule interval put in place, the logical date is going to indicate the time
at which it marks the start of the data interval, where the DAG run's start
date would then be the logical date + scheduled interval.
```

uranusjr reviewed (commented):
Some thoughts on wording. Generally looks nice, thanks for taking the time to put this in words! It’s inheritantly a difficult thing to explain!

howardyoo committed:
Update docs/apache-airflow/concepts/dags.rst

sounds good! I'll commit the changes.

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

howardyoo committed:
Update dags.rst

dag run does not have end date when it starts, but when it ends. Modified the description correctly.

howardyoo committed:
Update docs/apache-airflow/concepts/dags.rst

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

howardyoo committed:
Merge branch 'main' into patch-1
